feat: add sign up component and validation

yasminekharrat123 opened the pull request:
![image](https://github.com/user-attachments/assets/0a9aa3a4-3c88-4011-b4ad-c699ad95d07d)

Commit:
Created a new lib for the gameplay
Created the game main logic in a component game

Commit:
feat: add sign upcomponent and validation

skandertebo reviewed (changes requested):
to align with fields of Workos please use firstName and lastName instead of name

skandertebo committed:
feat(auth): add user registration and authentication

skandertebo committed:
Merge pull request #7 from skandertebo/skandertebo/add-password-auth-endpoint

feat(auth): add manual user registration and authentication with email and password

skandertebo commented on libs/frontend/type-it-up-landing/src/lib/signUp/signUp.component.ts:
```diff
@@ -0,0 +1,212 @@
+  Subject,
+  takeUntil,
+} from 'rxjs';
+// import { SignupService } from './signup.service';
```
do not comment out code

skandertebo commented on libs/frontend/type-it-up-landing/src/lib/signUp/signUp.component.ts:
```diff
@@ -0,0 +1,212 @@
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]/;
+  private readonly NAME_PATTERN = /^[a-zA-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$/;
+
+  // private signupService: SignupService
```
do not comment out code

skandertebo commented on libs/frontend/type-it-up-landing/src/lib/signUp/signUp.component.ts:
```diff
@@ -0,0 +1,212 @@
+            Validators.pattern(this.USERNAME_PATTERN),
+            this.noWhitespaceValidator(),
+          ],
+          // [this.usernameExistsValidator()],
```
do not comment out code

skandertebo commented on libs/frontend/type-it-up-landing/src/lib/signUp/signUp.component.ts:
```diff
@@ -0,0 +1,212 @@
+    };
+  }
+
+  // private usernameExistsValidator() {
+  //   return (control: AbstractControl): Observable<ValidationErrors | null> => {
+  //     return control.valueChanges.pipe(
+  //       debounceTime(500),
+  //       switchMap((username) =>
+  //         this.signupService.checkUsernameExists(username).pipe(
+  //           map((exists) => (exists ? { usernameExists: true } : null)),
+  //           first()
+  //         )
+  //       )
+  //     );
+  //   };
+  // }
```
remove commented code

Malek215 committed:
fixed naming conventions

skandertebo committed:
feat: add game module and update auth entities

skandertebo committed:
Merge pull request #8 from skandertebo/skandertebo/game

feat: add game module and update auth entities

MohamedRebai41 committed:
Merge pull request #4 from skandertebo/layout

type it up layout

skandertebo committed:
fix: testing fix

skandertebo committed:
Merge pull request #9 from skandertebo/skandertebo/game

fix: ci fix of previous pr in this branch

Commit:
feat: added font iceland

Commit:
feat: added generic button/toggle-field/toggle-list components

Commit:
feat: implemented game menu component
feat: implemented gameplay component
feat: improved game component

Commit:
feat: implemented gameService
feat: added wpm, accuracy and score
feat: implemented loading state for the button component

AminaGhannem committed:
create type it up profile lib

Commit:
Modified lib alias

Commit:
merged main, commented buggy auth code in layout

AminaGhannem committed:
feat: add user component

AminaGhannem committed:
feat: create accuracy and wpm chart

AminaGhannem committed:
fix: fix user components

AminaGhannem committed:
fix: some layout tweaks

MohamedRebai41 committed:
Merge pull request #10 from skandertebo/main_game

Main Game logic and UI

AminaGhannem committed:
fix: replace line chart with multi axis linr chart

AminaGhannem committed:
fix: uncomment

AminaGhannem committed:
fix: tweaks

AminaGhannem committed:
Merge branch 'main' of https://github.com/skandertebo/type-it-up into frontend/profile

skandertebo committed:
feat: wrote  gql queries and mutations for game and register and login in frontend

AminaGhannem committed:
fix: run the user component outside zonejs

MohamedRebai41 committed:
Merge pull request #12 from skandertebo/skandertebo/graphql-queries

feat: wrote  gql queries and mutations for game and register and logi…

skandertebo committed:
Merge pull request #11 from skandertebo/frontend/profile

Frontend/profile

Commit:
Merge branch 'main' of github.com:skandertebo/type-it-up into signup-component

Commit:
add routes and export components
